### Null Hypothesis (H0): Time series x does not Granger-cause time series y

### Alternative Hypothesis (HA): Time series x Granger-causes time series y

## ps5 buy online

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt

In [3]:
data=pd.read_csv('merged_data')

In [4]:
data.columns

Index(['Unnamed: 0', 'CALENDAR_DT', 'STD_SALES', 'DIG_SALES', 'TOTAL_SALES',
       'Demons Souls: (United States)',
       'playstation 5 buy online: (United States)',
       'ps5 buy online: (United States)',
       'sony ps5 buy online: (United States)',
       'Destruction AllStars: (United States)', 'Returnal: (United States)',
       'Ratchet & Clank: Rift Apart: (United States)',
       'Forspoken: (United States)', 'TOTAL_SALES_MA', 'PlayStation_MA',
       'TOTAL_SALES_SCALED', 'ps_5_MA', 'Forspoken_MA', 'Ratchet_MA'],
      dtype='object')

### x- TOTAL_SALES_SCALED
### y-ps5 buy online

In [11]:
from statsmodels.tsa.stattools import grangercausalitytests

#perform Granger-Causality test
grangercausalitytests(data[['TOTAL_SALES_SCALED', 'ps5 buy online: (United States)']], maxlag=[3])


Granger Causality
number of lags (no zero) 3
ssr based F test:         F=0.6354  , p=0.6001  , df_denom=22, df_num=3
ssr based chi2 test:   chi2=2.5127  , p=0.4730  , df=3
likelihood ratio test: chi2=2.4097  , p=0.4918  , df=3
parameter F test:         F=0.6354  , p=0.6001  , df_denom=22, df_num=3


{3: ({'ssr_ftest': (0.6353944688313293, 0.6001273469910806, 22.0, 3),
   'ssr_chi2test': (2.512696308560257, 0.4730009452100977, 3),
   'lrtest': (2.409746069563255, 0.4918229735275651, 3),
   'params_ftest': (0.6353944688313311, 0.6001273469910797, 22.0, 3.0)},
   array([[0., 0., 0., 1., 0., 0., 0.],
          [0., 0., 0., 0., 1., 0., 0.],
          [0., 0., 0., 0., 0., 1., 0.]])])}

### The F test statistic turns out to be 0.6354 and the corresponding p-value is 0.6001.

### Since the p-value is more than .05, we cannot reject the null hypothesis of the test and conclude that knowing that the sales scaled is not useful for predicting the future number of keywords.  (which is obvious)

## Reverse

### x-ps5 buy online
### y-TOTAL_SALES_SCALED

In [12]:
grangercausalitytests(data[['ps5 buy online: (United States)', 'TOTAL_SALES_SCALED']], maxlag=[3])


Granger Causality
number of lags (no zero) 3
ssr based F test:         F=7.4664  , p=0.0013  , df_denom=22, df_num=3
ssr based chi2 test:   chi2=29.5261 , p=0.0000  , df=3
likelihood ratio test: chi2=20.3631 , p=0.0001  , df=3
parameter F test:         F=7.4664  , p=0.0013  , df_denom=22, df_num=3


{3: ({'ssr_ftest': (7.4663582115512614, 0.0012627575367115337, 22.0, 3),
   'ssr_chi2test': (29.526052927498167, 1.7360699114799869e-06, 3),
   'lrtest': (20.36310905472385, 0.00014272828445117863, 3),
   'params_ftest': (7.466358211551263, 0.0012627575367115337, 22.0, 3.0)},
   array([[0., 0., 0., 1., 0., 0., 0.],
          [0., 0., 0., 0., 1., 0., 0.],
          [0., 0., 0., 0., 0., 1., 0.]])])}

### The F test statistic turns out to be 7.4664 and the corresponding p-value is 0.0013.
### Since the p-value is less than .05, we can reject the null hypothesis of the test and conclude that knowing that the number of keywords is  useful for predicting the future sales. 

## Forspoken

## Reverse

### x-Forspoken
### y-TOTAL_SALES_SCALED

In [16]:
grangercausalitytests(data[['Forspoken: (United States)', 'TOTAL_SALES_SCALED']], maxlag=[3])


Granger Causality
number of lags (no zero) 3
ssr based F test:         F=6.9581  , p=0.0018  , df_denom=22, df_num=3
ssr based chi2 test:   chi2=27.5160 , p=0.0000  , df=3
likelihood ratio test: chi2=19.3496 , p=0.0002  , df=3
parameter F test:         F=6.9581  , p=0.0018  , df_denom=22, df_num=3


{3: ({'ssr_ftest': (6.958063971113967, 0.0018269186584668178, 22.0, 3),
   'ssr_chi2test': (27.515980249405235, 4.588961413577411e-06, 3),
   'lrtest': (19.349600528990322, 0.00023146296318898862, 3),
   'params_ftest': (6.958063971113961, 0.0018269186584668256, 22.0, 3.0)},
   array([[0., 0., 0., 1., 0., 0., 0.],
          [0., 0., 0., 0., 1., 0., 0.],
          [0., 0., 0., 0., 0., 1., 0.]])])}

### The F test statistic turns out to be 6.9581 and the corresponding p-value is 0.0018.
### Since the p-value is less than .05, we can reject the null hypothesis of the test and conclude that knowing that the number of keywords is  useful for predicting the future sales. 

## Ratchet & Clank: Rift Apart

## Reverse

### x-Ratchet & Clank: Rift Apart
### y-TOTAL_SALES_SCALED

In [22]:
grangercausalitytests(data[['Ratchet & Clank: Rift Apart: (United States)', 'TOTAL_SALES_SCALED']], maxlag=[2])


Granger Causality
number of lags (no zero) 2
ssr based F test:         F=4.5006  , p=0.0214  , df_denom=25, df_num=2
ssr based chi2 test:   chi2=10.8014 , p=0.0045  , df=2
likelihood ratio test: chi2=9.2255  , p=0.0099  , df=2
parameter F test:         F=4.5006  , p=0.0214  , df_denom=25, df_num=2


{2: ({'ssr_ftest': (4.500569776380294, 0.021408276207513503, 25.0, 2),
   'ssr_chi2test': (10.801367463312706, 0.004513493868729055, 2),
   'lrtest': (9.225546463319006, 0.009924257843686888, 2),
   'params_ftest': (4.500569776380308, 0.021408276207513306, 25.0, 2.0)},
   array([[0., 0., 1., 0., 0.],
          [0., 0., 0., 1., 0.]])])}

### The F test statistic turns out to be 4.5006 and the corresponding p-value is 0.0214.
### Since the p-value is less than .05, we can reject the null hypothesis of the test and conclude that knowing that the number of keywords is  useful for predicting the future sales. 